### PR TITLE
various features

### PR DESCRIPTION
**Pull request by croqueGrec09**
includes commits for:
- ERMS-2906: code quality
- ERMS-3376: commenting out price item's startDate/endDate displays

### CODE REVIEW

File: grails-app/services/de/laser/ctrl/SubscriptionControllerService.groovy [hunk 1]
```diff
@@ -1187,18 +1187,21 @@ class SubscriptionControllerService {
             }
 
             Map query = filterService.getIssueEntitlementQuery(params, newSub)
-            List<IssueEntitlement> targetIEs = IssueEntitlement.executeQuery("select ie.id " + query.query, query.queryParams+[max: 5000, offset: 0])
-            List<IssueEntitlement> allIEs = subscriptionService.getIssueEntitlementsFixed(baseSub)
-            List<IssueEntitlement> notFixedIEs = subscriptionService.getIssueEntitlementsNotFixed(newSub)
+            List<IssueEntitlement> targetIEs = IssueEntitlement.executeQuery("select ie.id " + query.query, query.queryParams)
+            List<IssueEntitlement> allIEs = subscriptionService.getIssueEntitlementIDsFixed(baseSub)
```
Comment (Moetez16) [inline grails-app/services/de/laser/ctrl/SubscriptionControllerService.groovy:1191]: List<IssueEntitlement> allIEs != List<Long>

File: grails-app/services/de/laser/ctrl/SubscriptionControllerService.groovy [hunk 2]
```diff
@@ -1187,18 +1187,21 @@ class SubscriptionControllerService {
...
+            List<IssueEntitlement> notFixedIEs = subscriptionService.getIssueEntitlementIDsNotFixed(newSub)
```
Comment (Moetez16) [inline grails-app/services/de/laser/ctrl/SubscriptionControllerService.groovy:1192]: List<IssueEntitlement> allIEs != List<Long>

File: grails-app/services/de/laser/ctrl/SubscriptionControllerService.groovy [hunk 3]
```diff
@@ -1187,18 +1187,21 @@ class SubscriptionControllerService {
...
             result.countAllIEs = allIEs.size()
             result.countAllSourceIEs = sourceIEs.size()
             result.num_ies_rows = sourceIEs.size()
             //subscriptionService.getIssueEntitlementsFixed(baseSub).size()
             result.sourceIEIDs = sourceIEs
-            result.sourceIEs = IssueEntitlement.findAllByIdInList(sourceIEs.drop(result.offset).take(result.max))
-            result.targetIEs = IssueEntitlement.findAllByIdInList(targetIEs)
+            result.sourceIEs = sourceIEs ? IssueEntitlement.findAllByIdInList(sourceIEs.drop(result.offset).take(result.max)) : []
+            result.targetIEs = []
+            targetIEs.collate(32767).each {
```
Comment (Moetez16) [inline grails-app/services/de/laser/ctrl/SubscriptionControllerService.groovy:1202]: Why 32767?

Comment (croqueGrec09) [reply grails-app/services/de/laser/ctrl/SubscriptionControllerService.groovy:1202]: PostgreSQL has a parameter limit count to 32768 items

File: grails-app/services/de/laser/ctrl/SubscriptionControllerService.groovy [hunk 4]
```diff
@@ -2277,7 +2280,7 @@ class SubscriptionControllerService {
     }
 
     Map<String,Object> processRenewEntitlementsWithSurvey(SubscriptionController controller, GrailsParameterMap params) {
-        Map<String,Object> result = getResultGenericsAndCheckAccess(params, AccessService.CHECK_VIEW_AND_EDIT)
+        Map<String,Object> result = getResultGenericsAndCheckAccess(params, AccessService.CHECK_VIEW)
```
Comment (Moetez16) [inline grails-app/services/de/laser/ctrl/SubscriptionControllerService.groovy:2283]: Why only CHECK_VIEW?

File: grails-app/services/de/laser/ctrl/SubscriptionControllerService.groovy (hunk 3 above)
Comment (Moetez16) [reply grails-app/services/de/laser/ctrl/SubscriptionControllerService.groovy:1202]: ok

**Review by Moetez16 — APPROVED** [review]

**Pull request merged**